### PR TITLE
build: use new gateway images

### PR DESCRIPTION
**Pull request by bigglesandginger**
# Description

Use container images as built by the Uncharted-Auth CI
Additionally use Deployment versus Pod in Kubernetes files, Deployment allows for more options for future customization.

Resolves #(issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide reproduction instructions and list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by Nithos** [comment]
NOTE: This will replace the #126 PR.
Requires some testing by devs to verify it works as intended.

**Comment by YohannParis** [comment]
So I made sure to shut down k8 and docker, and delete all my images/containers.
Then I switch to this branch, and ran `./gateway.sh up`, it all worked well.
But I got the same error as before with keycloack being unhappy.

```shell
$ ./gateway.sh status
NAME                                    READY   STATUS                  RESTARTS      AGE
pod/gateway-httpd-666864895-srgvr       1/1     Running                 0             2m32s
pod/gateway-keycloak-6f44788d9f-b4qb2   0/1     Init:CrashLoopBackOff   4 (21s ago)   2m32s
pod/gateway-postgres-68f57bdc4c-4z454   0/1     ImagePullBackOff        0             2m33s

NAME                       TYPE           CLUSTER-IP      EXTERNAL-IP   PORT(S)          AGE
service/gateway-httpd      LoadBalancer   10.107.101.29   localhost     8078:30577/TCP   2m32s
service/gateway-keycloak   LoadBalancer   10.108.181.8    localhost     8079:32624/TCP   2m32s
service/gateway-postgres   ClusterIP      10.97.30.187    <none>        5432/TCP         2m33s
service/kubernetes         ClusterIP      10.96.0.1       <none>        443/TCP          34d

NAME                               DATA   AGE
configmap/gateway-httpd-config     1      2m32s
configmap/gateway-httpd-htdocs     3      2m32s
configmap/gateway-keycloak-realm   4      2m33s
configmap/kube-root-ca.crt         1      34d
```

To solve the problem I had to do `docker pull postgres:14.4-alpine`...
I think the `keycloak` image should have postgres as a dependency, or it should be part of the `gateway.sh up`.

File: kubernetes/local/gateway.sh [hunk 1]
```diff
@@ -3,21 +3,11 @@
...
-			docker pull docker.uncharted.software/auth/httpd-openidc:dev-1.0.1
-			docker pull docker.uncharted.software/auth/keycloak:dev-1.1.0
-    else
-        echo "not pulling latest images as Apple Silicon httpd-openidc has not been pushed to the repo"
-    fi
-
+		# Manually pulling images here, as the K8s sometimes timeout
+		docker pull ghcr.io/unchartedsoftware/httpd-openidc:0.1.2
+		docker pull ghcr.io/unchartedsoftware/keycloak:0.1.2
 		docker pull docker.uncharted.software/auth/terarium-theme:0.0.1
```
Comment (YohannParis) [inline kubernetes/local/gateway.sh:9]: Why this is not being pulled with `ghcr.io/darpa-askem/terarium-theme:latest`?

Comment (Nithos) [reply kubernetes/local/gateway.sh:9]: WIP: that package generated was a test, adding proper CI for this to happen now
additionally b/c its a `private` package will need everyone to login so adding documentation on how to do that

PR coming shortly

**Comment by YohannParis** [comment]
> Can you also run kubectl describe pod/gateway-keycloak...

```shell
$ kubectl describe pod/gateway-keycloak...
Error from server (NotFound): pods "gateway-keycloak..." not found
```

**Comment by Nithos** [comment]
Seems to work for me without the extra pull... unless I missed a deletion... will try again.
On another note should not be seeing a left green panel anymore?
image
Probably not related to this PR.

**Review by Nithos — COMMENTED** [review]
LGTM

Keep an eye out of any issues once devs start running it again.

**Comment by YohannParis** [comment]
> On another note should not be seeing a left green panel anymore?
> 
> Probably not related to this PR.

Indeed, the sidebar is not visible on the homepage anymore.

**Review by Nithos — APPROVED** [review]
LGTM

**Review by YohannParis — APPROVED** [review]
Okay, I will file this `docker pull postgres:14.4-alpine` problem in its own issue, that can be resolve another time.

**Pull request merged**